sklearn/linear_model/huber.py

```diff
@@ -253,11 +253,8 @@ def fit(self, X, y, sample_weight=None):
         X, y = check_X_y(
             X, y, copy=False, accept_sparse=['csr'], y_numeric=True,
             dtype=[np.float64, np.float32])
-        if sample_weight is not None:
-            sample_weight = np.array(sample_weight)
-            check_consistent_length(y, sample_weight)
-        else:
-            sample_weight = np.ones_like(y)
```

Comment:
I'm pretty sure this didn't produce sample weights with the expected dtype when `y` was an integer array.